sdk/src/wallet/operations/transaction/build_transaction.rs

```diff
@@ -48,6 +55,30 @@ where
             }
 
             inputs.push(Input::Utxo(UtxoInput::from(*input.output_id())));
+
+            if let Some(reward) = selected_transaction_data.mana_rewards.get(input.output_id()) {
+                mana_rewards += *reward;
+                context_inputs.insert(ContextInput::from(RewardContextInput::new(idx as _)?));
+                needs_commitment_context = true;
+            }
+        }
+
+        for output in selected_transaction_data
+            .outputs
+            .iter_mut()
+            .filter(|o| o.is_delegation())
+        {
```

Comment:
Honestly I don't think this whole block should be here. I get it that you need the latest_slot_commitment_id and here is the ideal place but later when we refactor the block options and have issuance set as first request, I think we should move this to its relevant place. But we'll see that later.

Reply:
okie